devicemodel/hw/pci/core.c

```diff
@@ -1059,6 +1059,79 @@ pci_emul_deinit(struct vmctx *ctx, struct pci_vdev_ops *ops, int bus, int slot,
 	}
 }
 
+static int
+pci_access_msi(struct pci_vdev *dev, int msi_cap, uint32_t *val, bool is_write)
+{
+	uint16_t msgctrl;
+	int rc, offset;
+
+	rc = pci_emul_find_capability(dev, PCIY_MSI, &offset);
+
+	if (msi_cap + offset > dev->prevcap) {
+		pr_err("%s: Msi capability length is out of msi length!\n", __func__);
+		return -1;
+	}
+	if (rc)
+		return -1;
+
```

Comment:
Keep it simple.
if (msi_cap > PCIR_MSI_PENDING) {
    pr_err("%s: invalid offset of msi capability.\n", __func__);
    return -1;
}

Reply:
Update the PR.